internal/transportmock/responder.go

```diff
@@ -0,0 +1,111 @@
+package transportmock
+
+import (
+	"bytes"
+	"encoding/json"
+	"io"
+	"net/http"
+)
+
+type Responder func(req *http.Request) (*http.Response, error)
+
+func HTTPResponse(status int, header *map[string][]string, body interface{}, cb func(*http.Request)) Responder {
+	return func(req *http.Request) (*http.Response, error) {
+		var b io.Reader
+		if s, ok := body.(string); ok {
+			b = bytes.NewBufferString(s)
+		} else {
+			s, _ := json.Marshal(body)
+			b = bytes.NewBuffer(s)
+		}
+		if cb != nil {
+			cb(req)
+		}
+		if header == nil {
+			header = &map[string][]string{}
+		}
+		return httpResponse(status, *header, b, req), nil
+	}
+}
+
+func RESTResponse(body interface{}, cb func(map[string]interface{})) Responder {
```

Comment:
I think having `HTTPResponse` and `RESTResponse` to be a bit confusing. Open to better naming here.

Reply:
Any less confusing if you call `RESTResponse` a `JSONResponse`? Afterall, it's just the response itself which happens to be JSON. Perhaps `HTTPResponse` could be `RAWResponse` and uses for non-JSON payloads - perhaps with a content-type? More focus on the payload than how it was retrieved.